src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1323 @@
+        }
+
+        // 0x00457AF3
+        static bool orderByProduction(const openloco::industry& lhs, const openloco::industry& rhs)
+        {
+            auto industryObj = objectmgr::get<industry_object>(lhs.object_id);
+            auto lhsVar = 0;
+
+            if ((industryObj->produced_cargo_type[0] & industryObj->produced_cargo_type[1]) != 0xFF)
+            {
+                lhsVar = lhs.produced_cargo_transported[0];
+                if (industryObj->produced_cargo_type[1] != 0xFF)
+                {
+                    lhsVar = lhs.produced_cargo_transported[1];
+                    if (industryObj->produced_cargo_type[0] != 0xFF)
+                    {
+                        lhsVar += lhs.produced_cargo_transported[0];
+                        lhsVar = lhsVar >> 1;
+                    }
+                }
+            }
+
+            industryObj = objectmgr::get<industry_object>(rhs.object_id);
+            auto rhsVar = 0;
+
+            if ((industryObj->produced_cargo_type[0] & industryObj->produced_cargo_type[1]) != 0xFF)
+            {
+                rhsVar = rhs.produced_cargo_transported[0];
+                if (industryObj->produced_cargo_type[1] != 0xFF)
+                {
+                    rhsVar = rhs.produced_cargo_transported[1];
+                    if (industryObj->produced_cargo_type[0] != 0xFF)
+                    {
+                        rhsVar += rhs.produced_cargo_transported[0];
+                        rhsVar = rhsVar >> 1;
+                    }
+                }
+            }
+            return rhsVar < lhsVar;
+        }
+
+        // 0x00457A52, 0x00457A9F, 0x00457AF3
+        static bool getOrder(const SortMode mode, openloco::industry& lhs, openloco::industry& rhs)
+        {
+            switch (mode)
+            {
+                case SortMode::Name:
+                    return orderByName(lhs, rhs);
+
+                case SortMode::Status:
+                    return orderByStatus(lhs, rhs);
+
+                case SortMode::Production:
+                    return orderByProduction(lhs, rhs);
+            }
+
+            return false;
+        }
+
+        // 0x00457991
+        static void updateIndustryList(window* self)
+        {
+            auto edi = -1;
+
+            auto i = -1;
+
+            for (auto& industry : industrymgr::industries())
+            {
+                i++;
+                if (industry.empty())
+                    continue;
+
+                if ((industry.flags & industry_flags::flag_02) != 0)
+                    continue;
+
+                if (edi == -1)
+                {
+                    edi = i;
+                    continue;
+                }
+
+                if (getOrder(SortMode(self->sort_mode), industry, *industrymgr::get(edi)))
+                {
+                    edi = i;
+                }
+            }
+
+            if (edi != -1)
+            {
+                bool dl = false;
```

Comment:
`shouldInvalidate`

Reply:
I mean that the variable should be called `shouldInvalidate`. Remove the extra invalidate and rename the variable.